sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1839,13 +1842,25 @@ class Analyzer(
                   }
                   AggregateExpression(agg, Complete, isDistinct, filter)
                 // This function is not an aggregate function, just return the resolved one.
-                case other =>
-                  if (isDistinct || filter.isDefined) {
-                    failAnalysis("DISTINCT or FILTER specified, " +
-                      s"but ${other.prettyName} is not an aggregate function")
-                  } else {
-                    other
+                case other if (isDistinct || filter.isDefined) =>
+                  failAnalysis("DISTINCT or FILTER specified, " +
+                    s"but ${other.prettyName} is not an aggregate function")
+                case e @ (_: StringTrimLeft | _: StringTrimRight) =>
```

Comment:
shall we keep the one-parameter `ltrim` and `rtrim`?

Reply:
Seems like one-parameter `ltrim` and `rtrim` have well defined semantic and are consistent with other databases. I'm OK to keep them, although users can do the same thing with the TRIM syntax.

Reply:
Ur, did you change your mind? This was due to your comment.

Reply:
Yea I just realized that ltrim/rtrim also has a one-parameter version, and should be kept like the one-parameter trim func.

Reply:
That was the reason I took a look this for a while. :)
Got it. I narrowed down the scope back.